src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -57,263 +57,326 @@ function fillColorArray(colors, offset, length, r, g, b) {
-    /* eslint-disable-next-line */
-    for (const id in coordinates.featureVertices) {
-        // extract contour coodinates and properties of one feature
-        const { contour, property } = extractFeature(coordinates, properties, id);
-        // get altitude and extrude amount from properties
-        const altitudeBottom = getAltitude(options, property, contour);
-        minAltitude = Math.min(minAltitude, altitudeBottom);
-        const extrudeAmount = getExtrude(options, property);
-        // altitudeTopFace is the altitude of the visible top face.
-        const altitudeTopFace = altitudeBottom + extrudeAmount;
-        // add vertices of the top face
-        coordinatesToExtrudedVertices(contour, altitudeTopFace, altitudeBottom, vertices, offset2);
-        // triangulate the top face
-        nbVertices = contour.length * 3;
-        const verticesTopFace = vertices.slice(offset2, offset2 + nbVertices);
-        const triangles = Earcut(verticesTopFace, null, 3);
+    // Build indices
+    if (!multiGeomAttributes) {
+        // Single polygon case
+        const isClockWise = THREE.ShapeUtils.isClockWise(
+            vertices.slice(geometry.contour.offset,
+                geometry.contour.offset +
+                geometry.contour.count).map(c => c.xyz()));
+
+        const holes = geometry.holes.map(h => h.offset);
+        const triangles = Earcut(geom.attributes.position.array.slice(
+            vertices.length * 3), holes, 3);
         for (const indice of triangles) {
-            indices.push(offset + indice);
+            indices.push(indice);
+        }
+        addExtrudedPolygonSideFaces(
+            indices,
+            vertices.length,
+            geometry.contour,
+            isClockWise);
+        if (holes.length > 0) {
+            for (let j = 0; j < geometry.holes.length; j++) {
+                addExtrudedPolygonSideFaces(
+                    indices,
+                    vertices.length,
+                    geometry.holes[j],
+                    isClockWise);
+            }
+        }
+    } else {
+        // Multi polygon case
+        const isClockWise = THREE.ShapeUtils.isClockWise(
+            vertices.slice(
+                multiGeomAttributes[0].offset + geometry[0].contour.offset,
+                multiGeomAttributes[0].offset +
+                geometry[0].contour.offset +
+                geometry[0].contour.count).map(c => c.xyz()));
+
+        for (let i = 0; i < geometry.length; i++) {
+            const holes = geometry[i].holes.map(h => h.offset);
+            // triangulate the top face
+            const start = vertices.length + multiGeomAttributes[i].offset + geometry[i].contour.offset;
+            const end = vertices.length + multiGeomAttributes[i].offset + multiGeomAttributes[i].count;
+            const triangles = Earcut(geom.attributes.position.array.slice(start * 3, end * 3),
+                holes,
+                3);
+            for (const indice of triangles) {
```

Comment:
If you try the `wfs` example, you'll get some error: `glDrawElements: attempt to access out of range vertices in attribute 0`. This is caused by the `indices` array being false, and from what I've seen it is only when `triangles` is empty at this stage.